tooling/nargo/src/errors.rs

```diff
@@ -60,6 +61,9 @@ pub enum ExecutionError {
 
     #[error(transparent)]
     SolvingError(#[from] OpcodeResolutionError),
+
+    #[error("Execution halted")]
+    Halted,
```

Comment:
This seems like a bit of a code smell to me. This seems like we're overloading the errors rather than using a more suitable return type.